aten/src/ATen/SparseTensorUtils.h

```diff
@@ -122,4 +127,24 @@ inline LongTensor flatten_indices_by_dims(const LongTensor& indices, const IntLi
   return new_indices;
 }
 
+// binary search for matching of a[a_i] at b, and b has size = b_nnz
```

Comment:
I find it always helpful to say when binary searching, if the element is not found, whether or not we return the left or the right of the slot in question.